docs/customizing_deis/choosing-a-scheduler.rst

```diff
@@ -122,16 +122,50 @@ The Swarm Scheduler is now active. Commands such as ``deis destroy`` or
 To monitor Swarm Scheduler operations, watch the logs of the swarm-manager
 component, or spy on Docker events directly on the swarm-manager machine:
 
+Kubernetes Scheduler
+--------------------
+
+.. important::
+
+    The Kubernetes Scheduler is a technology preview and is not recommended for
+    production use.
+
+`Kubernetes`_ orchestration system for docker containers:
+
+    Kubernetes provides APIs to manage, deploy and scale docker containers. Kubernetes deploys docker containers as `pods`_,
+    unique entity across a cluster. Containers inside a pod share the same namespaces. More information about `pods`_.
+
+Kubernetes requires overlay network so that each pod can get a unique IP across the cluster which is achieved by using `flannel`_.
+Deis repository provides a new user-data file to bootstrap CoreOs machines on the cluster with flannel.
+
+To test the Kubernetes Scheduler, first install and start the kubernetes components:
+
+.. code-block:: console
+
+    $deisctl install k8s && deisctl start k8s
+
+Then set the controller's ``schedulerModule`` to "k8s":
+
 .. code-block:: console
 
-    $ deisctl journal swarm-manager
-    $ docker -H 172.17.8.102:2395 events
-    2015-04-30T17:31 172.17.8.100:5000/hungry-variable:v5: (from  node:deis-01) pull
-    2015-04-30T17:31 172.17.8.100:5000/hungry-variable:v5: (from  node:deis-02) pull
-    2015-04-30T17:31 02a570: (from 172.17.8.100:5000/hungry-variable:v5 node:deis-01) create
-    2015-04-30T17:31 02a570: (from 172.17.8.100:5000/hungry-variable:v5 node:deis-01) start
-    2015-04-30T17:31 61e59c: (from 172.17.8.100:5000/hungry-variable:v5 node:deis-02) create
-    2015-04-30T17:31 61e59c: (from 172.17.8.100:5000/hungry-variable:v5 node:deis-02) start
+    $ deisctl config controller set schedulerModule=k8s
+
+The Kubernetes scheduler is now active. Commands such as ``deis destroy`` or
+``deis scale web=9`` will use kubernetes ApiServer to manage app pods.
+
+Deis creates a `replication controller`_ to manage pods and a `service`_ which acts as a proxy and routes traffic to the pods associated with the App.
+A new release uses rolling deploy mechanism where in we create and replace the pod with the new release one by one until all the pods are replaced properly or rollback to the older release
+if any error occurs which differs from the existing schedulers where we create all the containers at a time and replace them with the old ones.
+
+.. note::
+
+    **Known Issues**
+
+    - Kubernetes ApiServer is not HA. If ApiServer is rescheduled it will reschedule all the kubernetes units.
+    - Kubernetes does resource based scheduling. Specifying limits will create reservation of the specified resource on the node.
+    - Installing flannel is not backwards compatible as it changes Docker bridge which requires restarting
```

Comment:
@carmstrong  we are mentioning installing flannel is not back wards compatible. This implies that coreOS with flannel requires new cluster. Let me know if you want this to be rephrased or add a note saying kubernetes installation requires new coreOS cluster